Xamarin.Essentials/Screenshot/Screenshot.macos.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using ImageIO;
+using MobileCoreServices;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Screenshot
+    {
+        internal static bool PlatformIsCaptureSupported => true;
+
+        static Task<ScreenshotResult> PlatformCaptureAsync()
+        {
+            using var image = ApplicationServices.GetScreenshot();
+            var result = new ScreenshotResult(new NSImage(image, new CGSize(image.Width, image.Height)));
+            return Task.FromResult(result);
+        }
+    }
+
+    public partial class ScreenshotResult
+    {
+        readonly NSImage uiImage;
+
+        internal ScreenshotResult(NSImage image)
+        {
+            uiImage = image;
+
+            Width = (int)image.Size.Width;
+            Height = (int)image.Size.Height;
+        }
+
+        internal Task<Stream> PlatformOpenReadAsync(ScreenshotFormat format)
+        {
+            return Task.FromResult<Stream>(new ImageTypeStream(uiImage.CGImage, format));
```

Comment:
Do we need the stream type? What happens if we cache the data object, and then call as stream multiple times?

The GC will collect the image and data.

Reply:
We need it to not have to dispose ScreenshotResult and to be able to call PlatformOpenReadAsync multiple times